src/ripple/app/tx/impl/LoadManifest.cpp

```diff
@@ -0,0 +1,138 @@
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/tx/impl/LoadManifest.h>
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/app/misc/Manifest.h>
+#include <ripple/basics/Log.h>
+#include <ripple/protocol/digest.h>
+#include <ripple/protocol/Feature.h>
+#include <ripple/protocol/Indexes.h>
+#include <ripple/protocol/TxFlags.h>
+
+namespace ripple {
+
+std::uint64_t
+LoadManifest::calculateBaseFee (
+    ReadView const& view,
+    STTx const& tx)
+{
+    // Since this object will be perpetually stored on the ledger, we want to
+    // charge a fee that is enough to offset the cost that doing so imposes on
+    // everyone; we could use the manifest size as a proxy, but it's probably
+    // easier (for everyone) if we simply charge one incremental reserve.
+    return view.fees().increment * view.fees().units / view.fees().base;
+}
+
+NotTEC
+LoadManifest::preflight (PreflightContext const& ctx)
+{
+    if (! ctx.rules.enabled(featureOnLedgerManifests))
+        return temUNKNOWN;
+
+    if (ctx.tx.getFlags() & tfLoadManifestMask)
+        return temINVALID_FLAG;
+
+    auto const ret = preflight1 (ctx);
+
+    if (!isTesSuccess (ret))
+        return ret;
+
+    auto const mb = ctx.tx[sfManifest];
+
+    if (mb.size() < 32 || mb.size() > 768)
+        return temMANIFEST_MALFORMED;
+
+    auto const m = deserializeManifest(ctx.tx[sfManifest]);
+
+    if (!m)
+        return temMANIFEST_MALFORMED;
+
+    // Existing code will not deserialize a manifest with a domain name
+    // that is longer than maxDomainLength. However, such a change, if
+    // ever made, would be transaction breaking if deployed without an
+    // amendment. This extra check here protects from this unlikely
+    // scenario.
+    if (m->domain.size() > maxDomainLength)
+        return telBAD_DOMAIN;
+
+    return preflight2 (ctx);
+}
+
+TER
+LoadManifest::preclaim(PreclaimContext const& ctx)
+{
+    if (! ctx.view.rules().enabled(featureOnLedgerManifests))
+        return temDISABLED;
+
+    auto m = deserializeManifest(ctx.tx[sfManifest]);
+
+    if (!m)
+        return temMANIFEST_MALFORMED;
+
+    if (!m->verify())
+        return tecMANIFEST_BAD_SIGNATURE;
+
+    return tesSUCCESS;
```

Comment:
I'd rather have the server charge when it is forced to do work that amounts to nothing because of bad input. I rewrote this code a bit, but we are still overly eager to deserialize the input. I may do another [FOLD] commit that reduces the number of calls by one.

Thanks!

Reply:
Oh, I see. It's not about the `view`, it's about returning the `tec`, which you can't do from `preflight`. That makes sense. I peeked at the latest changes, and agree with where they went.